composer.json

```
@@ -16,7 +16,7 @@
 	],
 	"require": {
 		"php": "^7.2 || ^8.0",
-		"smarty/smarty": "^3",
+		"smarty/smarty": "^4",
```

This can be

```suggestion
		"smarty/smarty": "^3 || ^4",
```

which would only install Smarty 4 if it's compatible with the rest of the stack, and not change anything otherwise.